s3_management/manage.py

```diff
@@ -348,10 +351,13 @@ def from_S3(cls: Type[S3IndexType], prefix: str) -> S3IndexType:
                 for pattern in ACCEPTED_SUBDIR_PATTERNS
             ]) and obj.key.endswith(ACCEPTED_FILE_EXTENSIONS)
             if is_acceptable:
+                response = obj.meta.client.head_object(Bucket=BUCKET.name, Key=obj.key, ChecksumMode="ENABLED")
+                sha256 = response.get("ChecksumSHA256")
                 sanitized_key = obj.key.replace("+", "%2B")
-                objects.append(sanitized_key)
+                objects.append((sanitized_key, sha256))
```

Comment:
This still has `objects` as a list. I don't see it being converted to a `dict` anywhere?

I think you can just change it on line 343 to a `dict` and here to `objects[sanitized_key] = sha256`